compiler/util/OwnershipFlowManager.cpp

```diff
@@ -788,11 +784,73 @@ void OwnershipFlowManager::autoCopyForSimpleAssignment(CallExpr* call,
 
     processBitwiseCopy(lhse, prod, live, cons);
   }
+}
+
+// If the RHS is a call to a function, no autocopy is needed because calls
+// are uniformly treated as returning an owned value.  If the thing being
+// copied into the RVV is owned, we don't need to insert an autoCopy.
+// But that still leaves the case where a CallExpr on the RHS returns
+// something that is unowned.
```

Comment:
I'm not as confident on my grasp of when autocopy calls are and aren't required, so I feel hesitant to sign off on this justification.  I would like @mppf to verify that what this comment describes is true, but other than that the code looks fine.